src/test/java/org/junitpioneer/internal/AllElementsAreEqual.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2016-2022 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.internal;
+
+import java.util.stream.StreamSupport;
+
+import org.assertj.core.api.Condition;
+
+public final class AllElementsAreEqual extends Condition<Iterable<?>> {
```

Comment:
This is way too fancy. Delete this class and use something simpler, e.g.:
```java
List<Path> paths = TwoTestMethodsWithSharedSameNameTempDirParameterTestCases.recordedPaths;
assertThat(paths)
    .hasSize(2)
    .allSatisfy(path -> assertThat(path).isEqualTo(paths.get(0)))
    .allSatisfy(path -> assertThat(path).doesNotExist());
```

Reply:
Sounds reasonable. Fixed in https://github.com/junit-pioneer/junit-pioneer/pull/491/commits/3cbab7b21cd2aa7961f6b8e0814e2a4d08abd816.